system_metrics_collector/src/system_metrics_collector/metrics_message_publisher.hpp

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SYSTEM_METRICS_COLLECTOR__METRICS_MESSAGE_PUBLISHER_HPP_
+#define SYSTEM_METRICS_COLLECTOR__METRICS_MESSAGE_PUBLISHER_HPP_
+
+#include <chrono>
+#include <string>
+
+#include "builtin_interfaces/msg/time.hpp"
+#include "metrics_statistics_msgs/msg/metrics_message.hpp"
+
+#include "../moving_average_statistics/types.hpp"
+
+namespace system_metrics_collector
+{
+
+/**
+ * Simple class to facilitate publishing messages containing statistics data
+ */
+class MetricsMessagePublisher
+{
+public:
+  static constexpr const std::chrono::milliseconds INVALID_PUBLISH_WINDOW =
+    std::chrono::milliseconds(0);
```

Comment:
Let's avoid 0 -> -1

Reply:
We might just want to define in tests because I believe it's the only place this is used.

Reply:
We're going in circles on this. Can we punt on this?

Reply:
No, all review comments need to be resolved. Let's go with Devin's suggestions.

Reply:
Okay, then I believe I should remove all the `publish_timer_` along with this constant, since it's not functional right now anyway. I'll put it back in a future PR.

Reply:
> Okay, then I believe I should remove all the `publish_timer_` along with this constant, since it's not functional right now anyway. I'll put it back in a future PR.

As long as we can continue with working unit tests, as the tests rely on an option to disable publishing.